### PR TITLE
dispatcher options include env, not just envid

### PR DESCRIPTION
**Pull request by itayd**
- dispatcher option receives env, not just env id. this allows for integrations to direct events specifically to certain envs without knowing their id, without using tokens.
- http integration use of ^.
- token is not mandated in manifest and cmd.
- fix some old StrictParseXID to Strict(ParseXID).

### CODE REVIEW

File: integrations/http/server.go [hunk 1]
```diff
@@ -23,16 +29,21 @@ type HTTPHandler struct {
...
 	h := HTTPHandler{dispatcher: d, logger: l}
-	mux.Handle(routePrefix, http.StripPrefix(routePrefix, h))
+	mux.Handle(routePrefix("{ns}")+"*", h)
 }
 
 func (h HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	l := h.logger.With(zap.String("url", r.URL.String()))
 
-	l.Info("Incoming request")
+	l.Info("incoming request")
```
Comment (daabr) [inline integrations/http/server.go:38]: We need to standardize on logging conventions throughout the code - levels, text, and args - but not a concern for this PR.

File: internal/backend/dispatcher/eventsworkflow.go [hunk 2]
```diff
@@ -24,6 +25,62 @@ func (d *dispatcher) createEventRecord(ctx context.Context, eventID sdktypes.Eve
...
+	if env == "" {
+		return sdktypes.InvalidEnvID, nil
+	}
+
+	parts := strings.Split(env, "/")
+	switch len(parts) {
+	case 1:
+		if sdktypes.IsEnvID(parts[0]) {
+			return sdktypes.ParseEnvID(parts[0])
+		}
```
Comment (daabr) [inline internal/backend/dispatcher/eventsworkflow.go]: It's hard to follow the logic in this branch, i.e. need to skip to the end of the function. Why not call `return` here?

Also, why do we even support IDs? They're a purely internal implementation detail which users should not be exposed to or rely upon. It should be just names.

File: internal/backend/dispatcher/eventsworkflow.go [hunk 3]
```diff
@@ -24,6 +25,62 @@ func (d *dispatcher) createEventRecord(ctx context.Context, eventID sdktypes.Eve
 	return nil
 }
 
+func (d *dispatcher) resolveEnv(ctx context.Context, env string) (envID sdktypes.EnvID, err error) {
```
Comment (daabr) [inline internal/backend/dispatcher/eventsworkflow.go]: Not a single comment in ~50 LOC. Please add comments inside this function that explain its flow.

File: internal/backend/dispatcher/eventsworkflow.go [hunk 4]
```diff
@@ -24,6 +25,62 @@ func (d *dispatcher) createEventRecord(ctx context.Context, eventID sdktypes.Eve
...
+
+	parts := strings.Split(env, "/")
+	switch len(parts) {
+	case 1:
+		if sdktypes.IsEnvID(parts[0]) {
+			return sdktypes.ParseEnvID(parts[0])
+		}
+	case 2:
+		var pid sdktypes.ProjectID
+		if sdktypes.IsProjectID(parts[0]) {
```
Comment (daabr) [inline internal/backend/dispatcher/eventsworkflow.go]: Same as above - I don't think we should support IDs in URLs.

File: tests/system/testdata/connections/create.txtar [hunk 5]
```diff
@@ -4,32 +4,28 @@ return code == 0
 output equals 'project_id: prj_00000000000000000000000001'
 
 # Negative test: create connection without name.
-ak connection create --project my_project --integration http --connection-token my_url_path
+ak connection create --project my_project --integration http --connection-token tok
```
Comment (daabr) [inline tests/system/testdata/connections/create.txtar]: Please use a human-friendly string like `"my_token"` instead of the meaningless abbreviation `"tok"` (in all instances below too).

File: tests/system/testdata/connections/create.txtar [hunk 6]
```diff
@@ -4,32 +4,28 @@ return code == 0
...
 output equals 'Error: required flag(s) "integration" not set'
 return code == 1
 
-ak connection create my_connection --project my_project --integration http
-output equals 'Error: required flag(s) "connection-token" not set'
-return code == 1
-
 # Create connection.
-ak connection create my_connection --project my_project --integration http --connection-token my_url_path
+ak connection create my_connection --project my_project --integration http --connection-token tok
```
Comment (daabr) [inline tests/system/testdata/connections/create.txtar]: Please add another test for creating a connection without a token.

**Review by daabr — COMMENTED** [review]
I want to see your comments in the `resolveEnv` function to understand it better, and then I'll approve.

File: internal/backend/dispatcher/eventsworkflow.go (hunk 4 above)
Comment (itayd) [reply internal/backend/dispatcher/eventsworkflow.go]: why not?

File: internal/backend/dispatcher/eventsworkflow.go (hunk 2 above)
Comment (itayd) [reply internal/backend/dispatcher/eventsworkflow.go]: simplified now. no ids.

File: internal/backend/dispatcher/eventsworkflow.go (hunk 3 above)
Comment (itayd) [reply internal/backend/dispatcher/eventsworkflow.go]: fixed

File: internal/backend/dispatcher/eventsworkflow.go (hunk 4 above)
Comment (itayd) [reply internal/backend/dispatcher/eventsworkflow.go]: scratch that, you're right.

File: tests/system/testdata/connections/create.txtar (hunk 5 above)
Comment (itayd) [reply tests/system/testdata/connections/create.txtar]: done

**Review by daabr — APPROVED** [review]
